### PR TITLE
Implement Formatter support

### LINKED ISSUE #37: Add option to autofix on save
I would like to have the option to autofix on save, from what I've seen it doesn't look like this is currently possible (in this case, an enhancement request). Otherwise, if it is possible could you update the documentation to show how to configure this?

Thanks!

### PR DESCRIPTION
**Pull request by xlc**
Fixes #37 
To enable autofix on save, those options need to be turn on
```
"xo.enable": true,
"xo.format.enable": true,
"editor.formatOnSave": true
```

Known issue:
It only does a single fix pass which means not all the issues are fixed. Similar to https://github.com/Microsoft/vscode-eslint/issues/154

This PR is inspired by https://github.com/Microsoft/vscode-eslint/pull/466

I already published this change in my fork extension which you can give it a try: https://marketplace.visualstudio.com/items?itemName=bryan-chen.linter-xo-2

### CODE REVIEW

File: xo/package.json [hunk 1]
```diff
@@ -64,11 +71,11 @@
     "postinstall": "npm run update-vscode"
   },
   "dependencies": {
-    "vscode-languageclient": "^3.5.0"
+    "vscode-languageclient": "^4.1.3"
```
Comment (SamVerschueren) [inline xo/package.json]: Can you send these package upgrades in a separate PR?

File: xo/extension.ts [hunk 2]
```diff
@@ -15,18 +26,45 @@ namespace AllFixesRequest {
 	export const type = new RequestType<AllFixesParams, AllFixesResult, void, void>('textDocument/xo/allFixes');
 }
 
+const defaultLanguages = ['javascript', 'javascriptreact', 'typescript', 'typescriptreact'];
```
Comment (SamVerschueren) [inline xo/extension.ts]: Rename to `supportedLanguages`

**Comment by xlc** [comment]
Close in favour of #46
